Microsoft.Azure.Cosmos.Samples/Tools/FaultInjection/tests/FaultInjectionDirectModeTests.cs

```diff
@@ -1144,61 +1145,67 @@ private async Task Timeout_FaultInjectionServerErrorRule_ReadFeedTest()
 
             await this.Initialize(faultInjector, false);
 
-            try
+            string changeFeedContainerName = "changeFeedContainer-" + Guid.NewGuid().ToString();
+            ContainerProperties containerProperties = new ContainerProperties
             {
-                string changeFeedContainerName = "changeFeedContainer-" + Guid.NewGuid().ToString();
-                ContainerProperties containerProperties = new ContainerProperties
-                {
-                    Id = changeFeedContainerName,
-                    PartitionKeyPath = "/partitionKey"
-                };
-                if (this.database != null && this.container != null)
-                {
-                    Container? leaseContainer = await this.database.CreateContainerIfNotExistsAsync(containerProperties, 400);
+                Id = changeFeedContainerName,
+                PartitionKeyPath = "/partitionKey"
+            };
 
-                    ChangeFeedProcessor changeFeedProcessor = this.container.GetChangeFeedProcessorBuilder<JObject>(
-                        processorName: "FaultInjectionTest",
-                        onChangesDelegate: HandleChangesAsync)
-                        .WithInstanceName("test")
-                        .WithLeaseContainer(leaseContainer)
-                        .Build();
-                    await changeFeedProcessor.StartAsync();
-                    await Task.Delay(1000);
+            if (this.database != null && this.container != null)
+            {
+                JObject createdItem = JObject.FromObject(new { id = Guid.NewGuid().ToString(), Pk = Guid.NewGuid().ToString() });
+                ItemResponse<JObject>? itemResponse = await this.container.CreateItemAsync<JObject>(createdItem);
 
-                    JObject createdItem = JObject.FromObject(new { id = Guid.NewGuid().ToString(), Pk = Guid.NewGuid().ToString() });
+                readFeedRule.Enable();
 
-                    ItemResponse<JObject>? itemResponse = this.container != null
-                        ? await this.container.CreateItemAsync<JObject>(createdItem)
-                        : null;
-                    Assert.IsNotNull(itemResponse);
+                Container? leaseContainer = await this.database.CreateContainerIfNotExistsAsync(containerProperties, 400);
 
-                    readFeedRule.Enable();
+                ManualResetEvent changeFeedRan = new ManualResetEvent(false);
 
-                    await Task.Delay(5000);
-                }
-            }
-            finally
-            {
-                readFeedRule.Disable();
-            }
-        }
+                ChangeFeedProcessor changeFeedProcessor;
+                changeFeedProcessor = this.container.GetChangeFeedProcessorBuilder<JObject>(
```

Comment:
```suggestion
                ChangeFeedProcessor changeFeedProcessor = this.container.GetChangeFeedProcessorBuilder<JObject>(
```

And remove previous line